sdk/storage/azblob/blockblob/client.go

```diff
@@ -411,11 +412,17 @@ func (bb *Client) uploadFromReader(ctx context.Context, reader io.ReaderAt, actu
 		return uploadFromReaderResponse{}, errors.New("block limit exceeded")
 	}
 
+	urlparts, err := blob.ParseURL(bb.generated().Endpoint())
```

Comment:
Does this sound like an overhead? just wanted to print the blob name as well in the log otherwise its difficult to corelate for which blob we are printing the size and count.

Reply:
For log-only calculations that are expensive, you can wrap them in a `Should` check.  Here's an example.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/policy_logging.go#L98-L110

Reply:
add a should check here for upload event.